### ISSUE TITLE
Offset - optional params

### ISSUE THREAD

**Issue by pLeBlanc93** [Esri/local-government-desktop-addins#207]
The Offset method behaves differently with the following VALUEINFO:
- `Source` - fails to find a line
- `Source||` - finds the line and applies an `offsetVal` of 5

Only works with intersecting lines since 0 is treated as the `Neighborhood`.

The `Offset` and `Neighborhood` args could be made optional, resulting in the following scenarios:
- `Source` - find the intersecting line and populate the point location (not very useful, but possible)
- `Source|Offset` - find the intersecting line and populate the point location of a tangent offset
- `Source||Neighborhood` - find the line within `x` units and populate the point location if it were on the line
- `Source|Offset|Neighborhood` - find the line within `x` units and populate the point location if it were on the line + the tangent offset (current implementation)

In short, making the final two arguments optional gives useful functionality.
